### ISSUE TITLE
Discover works fine, but unable to run index.js with success

### ISSUE THREAD

**Issue by fucci** [voodootikigod/node-rolling-spider#74]
Colleague is running successful with Mac, I am running Ubuntu 14.04 with noble, temporal, rolling-spider, and bluetooth-hci-socket node modules. 

Discover showed my minidrone
1: RS_B075911 (e014b3783d48), RSSI -54

When I run index.js, just hangs

Any advice is appreciated, 
also any pointers on how to get debugging or logging on also appreciated.

Apologies if I sound dumb..... still getting up to speed with node

Thanks for this great piece of code/functionality

**Comment by roeku** [comment]
I had the same problem

https://github.com/voodootikigod/node-rolling-spider/issues/68

**Comment by fucci** [comment]
Thanks roeku, you mentioned, the following - so the BLE stops scanning, and it can then set up
this.ble.stopScanning();
- is this correct

Also, there are 2 more things I should mention
A. when I say the index.js hangs, I get the following coming up, which I think it means it stops
1: RS_B075911 (e014b3783d48), RSSI -36
B. assuming it does not matter where I run the index.js files

appreciate your help

**Comment by fucci** [comment]
Thanks roeku, you mentioned, the following - so the BLE stops scanning, and it can then set up
this.ble.stopScanning();

```
is this correct
```

Also, there are 2 more things I should mention
A. when I say the index.js hangs, I get the following coming up, which I think it means it stops

noble warning: unknown peripheral e014b3783d48

B. assuming it does not matter where I run the index.js files

appreciate your help

**Comment by roeku** [comment]
Go into lib/drone.js and add this line to 201
 `this.ble.stopScanning();`
under 
      `this.peripheral = peripheral;`

And then execute the index.js or another file in the eg folder. As far as I know this should resolve your issue.

**Comment by fucci** [comment]
yippee !!!!

that worked - many thanks

I am still a little confused, why I had to include that one line of code in
my version on ubuntu
- when my colleague on Mac did not

regardless - I am very happy
- Thank You

**Comment by beverts312** [comment]
Thanks @roeku !

**Comment by barackd222** [comment]
For us this solved the problem, but not after having to re-connect
Thank you

**Comment by Thyxx** [comment]
Edit: 
My bad, the Rolling Spider was paired with computer...
Remove the pairing, run the script and all work perfectly.

> Hi, 
> Same problem here:
> `noble warning: unknown peripheral eXXXXXXXXXXX`
> 
> Discover.js works fine but I can't run index.js
> Already add `this.ble.stopScanning();` but there is no difference.
> 
> Drone: Rolling-Spider
> Distrib: Debian 8
> 
> Thanks in advance.

**Comment by lynnaloo** [comment]
Is this still an issue?

**Comment by Nighthawk1901** [comment]
yes, it is. Even in the latest updates version it still shows. I have windows 10 and using rolling spider.

**Comment by Thyxx** [comment]
Problem solved...
I edited the message.

**Comment by Nighthawk1901** [comment]
I simply changed my built in Bluetooth...all you have to do in windows is use zadig, select intel wireless Bluetooth and replace the driver to WINusb. To change back, simply download the update for the driver from intel, run the program, and click repair, then restart your pc for it too work. I dont know if changing this however, can effect how well the drone operates.

**Comment by Nighthawk1901** [comment]
And i just tried. again. Still not working, same error "noble warning: unknown peripheral exxxxxxxxxxxx". So i hit ctrl c on nodejs cmd prompt, and write the cmd again "node rs_test.js". The error doesn't come up, and the drone takes of and does what it was meant to. When it finishes, I write the same cmd again, and I get "unknown handle 3585 disconnected". After that i get the same noble warning error, and the drone doesn't fly again. Only flies once, then wont fly again until i reboot my pc and the drone. Even then, it's luck. What do i do?
